src/ripple/shamap/impl/SHAMap.cpp

```diff
@@ -1373,7 +1384,20 @@ SHAMap::invariants() const
     auto node = root_.get();
     assert(node != nullptr);
     assert(!node->isLeaf());
+    SharedPtrNodeStack stack;
+    for (auto leaf = peekFirstItem(stack); leaf != nullptr;
+         leaf = peekNextItem(leaf->peekItem()->key(), stack))
+        ;
```

Comment:
This is common "empty loop" syntax.  I find it more readable than putting the ; at the end of the line on the for-loop.